src/protocols/displayport.stanza

```diff
@@ -0,0 +1,220 @@
+doc: \<DOC>
+@brief DisplayPort Protocol
+
+DisplayPort is a serial protocol supporting high speed links for video+audio
+@see https://en.wikipedia.org/wiki/DisplayPort
+
+This functions and definitions in this file support defining DisplayPort
+connections between sources and receivers on a printed circuit board.
+
+## DisplayPort Blocking Capacitors
+
+The DisplayPort specification calls for AC coupling for the data lanes. This is typically
+achieved using a blocking capacitor. When connecting two active components, this typically
+means blocking caps from `Tx -> Rx` on both sides of the link. When connecting an
+active component to a passive component, this typically means adding the blocking caps only on
+the `Tx -> Rx` side of the link.
+
+The functions in this module allow you to pass a blocking capacitor as an `Instantiable`.
+This component will get instantiated for each leg of the diff-pair. These functions handle
+the topology configuration, but the user needs to set a `pin-model` statement on the capacitor component.
+
+<DOC>
+#use-added-syntax(jitx)
+defpackage jsl/protocols/displayport:
+  import core
+  import jitx
+  import jitx/commands
+
+  import jsl/bundles
+  import jsl/errors
+  import jsl/si/helpers
+  import jsl/si/couplers
+  import jsl/pin-assignment
+
+
+doc: \<DOC>
+@brief DisplayPort Generation enums
+This is a fixed list of possible generation definitions for DisplayPort
+<DOC>
+
+public pcb-enum jsl/protocols/displayport/DPVersion:
+  DP1p0  ; 10.80 Gbit/s / 4 lanes =>  2.7 Gbit/s / 10 UI =>  270 MHz clk
+  DP1p1  ; 10.80 Gbit/s / 4 lanes =>  2.7 Gbit/s / 10 UI =>  270 MHz clk
+  DP1p2  ; 21.60 Gbit/s / 4 lanes =>  5.4 Gbit/s / 10 UI =>  540 MHz clk
+  DP1p2a ; 21.60 Gbit/s / 4 lanes =>  5.4 Gbit/s / 10 UI =>  540 MHz clk
+  DP1p3  ; 32.40 Gbit/s / 4 lanes =>  8.1 Gbit/s / 10 UI =>  810 MHz clk
+  DP1p4  ; 32.40 Gbit/s / 4 lanes =>  8.1 Gbit/s / 10 UI =>  810 MHz clk
+  DP1p4a ; 32.40 Gbit/s / 4 lanes =>  8.1 Gbit/s / 10 UI =>  810 MHz clk
+  DP2p0  ; 80.00 Gbit/s / 4 lanes => 20.0 Gbit/s / 10 UI => 2000 MHz clk
+  DP2p1  ; 80.00 Gbit/s / 4 lanes => 20.0 Gbit/s / 10 UI => 2000 MHz clk
+  DP2p1a ; 80.00 Gbit/s / 4 lanes => 20.0 Gbit/s / 10 UI => 2000 MHz clk
+
+doc: \<DOC>
+@brief DisplayPort Bundle
+The Data Bundle consists of 4 lane pairs (1 RX or TX pair set = 1 lane)
+The bundle also constains an AUX channel (differential pair) and a HPD pin
+<DOC>
+
+public pcb-bundle displayport :
+  name = "DisplayPort"
+  description = "DisplayPort Digital Display Interface"
+  port aux  : diff-pair
+  port lane : diff-pair[4]
+  port hpd  : pin ; hot plug detect
+
+
+defn constrain-ch (sw:Toleranced, ml:Double, a:JITXObject, b:JITXObject) :
```

Comment:
Duplicate - lets create a function in `si/helpers.stanza` that implements this concept.

Reply:
Good idea. Will do.